eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -237,177 +404,591 @@ object Transactions {
+            txOut = TxOut(amount, localFinalScriptPubKey) :: Nil,
+            lockTime = htlc.cltvExpiry.toLong
+          )
+          val unsignedTx = ClaimHtlcTimeoutTx(input, tx, htlc.paymentHash, htlc.id, htlc.cltvExpiry)
+          skipTxIfBelowDust(unsignedTx, dustLimit, () => unsignedTx.sign(commitKeys, commitmentFormat))
+        case None => Left(OutputNotFound)
+      }
+    }
+  }
+
+  /** This transaction claims our anchor output in either the local or remote commitment, to CPFP and get it confirmed. */
   case class ClaimAnchorOutputTx(input: InputInfo, tx: Transaction, confirmationTarget: ConfirmationTarget) extends ReplaceableTransactionWithInputInfo {
     override val desc: String = "local-anchor"
 
-    def addSigs(localSig: ByteVector64): ClaimAnchorOutputTx = input match {
-      case InputInfo.SegwitInput(_, _, redeemScript) =>
-        val witness = witnessAnchor(localSig, redeemScript)
-        copy(tx = tx.updateWitness(0, witness))
-      case _: InputInfo.TaprootInput => ???
+    def sign(fundingKey: PrivateKey, commitKeys: LocalCommitmentKeys, commitmentFormat: CommitmentFormat, extraUtxos: Map[OutPoint, TxOut]): ClaimAnchorOutputTx = {
+      commitmentFormat match {
+        case DefaultCommitmentFormat | _: AnchorOutputsCommitmentFormat => sign(fundingKey, commitmentFormat, extraUtxos)
+      }
+    }
+
+    def sign(fundingKey: PrivateKey, commitKeys: RemoteCommitmentKeys, commitmentFormat: CommitmentFormat, extraUtxos: Map[OutPoint, TxOut]): ClaimAnchorOutputTx = {
+      commitmentFormat match {
+        case DefaultCommitmentFormat | _: AnchorOutputsCommitmentFormat => sign(fundingKey, commitmentFormat, extraUtxos)
+      }
+    }
+
+    private def sign(anchorKey: PrivateKey, commitmentFormat: CommitmentFormat, extraUtxos: Map[OutPoint, TxOut]): ClaimAnchorOutputTx = {
+      val witness = commitmentFormat match {
+        case DefaultCommitmentFormat | _: AnchorOutputsCommitmentFormat =>
+          val redeemScript = Script.write(anchor(anchorKey.publicKey))
+          val sig = sign(anchorKey, sighash(TxOwner.Local, commitmentFormat), RedeemInfo.P2wsh(redeemScript), extraUtxos)
+          witnessAnchor(sig, redeemScript)
+      }
+      copy(tx = tx.updateWitness(inputIndex, witness))
+    }
+  }
+
+  object ClaimAnchorOutputTx {
+    def redeemInfo(fundingKey: PrivateKey, commitKeys: CommitmentPublicKeys, commitmentFormat: CommitmentFormat): RedeemInfo = {
+      commitmentFormat match {
+        case DefaultCommitmentFormat | _: AnchorOutputsCommitmentFormat =>
+          val redeemScript = Script.write(anchor(fundingKey.publicKey))
+          RedeemInfo.P2wsh(redeemScript)
+      }
+    }
+
+    def createUnsignedTx(fundingKey: PrivateKey, commitKeys: CommitmentPublicKeys, commitTx: Transaction, confirmationTarget: ConfirmationTarget, commitmentFormat: CommitmentFormat): Either[TxGenerationSkipped, ClaimAnchorOutputTx] = {
```

Comment:
Same as above, we need 2 `createUnsignedTx` methods, one for `LocalCommitmentKeys` and the other for `RemoteCommitmentKeys`